### PR TITLE
Delete SharedBlob and SharedBlobs

### PR DESCRIPTION
**Pull request by sagar-solana**
#### Problem

SharedBlob and SharedBlobs were unnecessary sync wrappers around Blobs. 

#### Summary of Changes

Deleted SharedBlob and SharedBlobs

### CODE REVIEW

**Comment by sagar-solana** [comment]
@mark-solana, @carllin, please help make sure I'm not missing anything in blocktree and erasure. Specifically places where sharedblobs were being passed in and modified.

**Comment by rob-solana** [comment]
it'd be nice if this was parked until erasure were fixed, likely tomorrow

**Comment by sagar-solana** [comment]
Sure. Sucks that I've had to rebase it both ways lol

**Comment by rob-solana** [comment]
> Sure. Sucks that I've had to rebase it both ways lol

ok, then: go for it.  it'll likely end up being a lot less code for erasure stuff, too

**Comment by sagar-solana** [comment]
Depends on whether or not I can figure out the coverage failure in time. Probably won't look at it until tomorrow.

**Comment by mark-solana** [comment]
waiting on #3889 to merge this

File: core/src/blocktree.rs [hunk 1]
```diff
@@ -2792,15 +2753,14 @@ pub mod tests {
                         // between 1-4 data blobs are missing
                         if rng.gen() {
                             blocktree
-                                .write_shared_blobs(erasure_set.data)
+                                .write_blobs(erasure_set.data)
```
Comment (rob-solana) [inline core/src/blocktree.rs]: write_blobs() move them, yes?

**Comment by rob-solana** [comment]
I think we need to look closely at when we're copying a big bunch of data in this PR.  It's not obvious in a 5-minute review where we might be opening ourselves up to memmove() being big in a flamegraph

File: core/src/blocktree.rs (hunk 1 above)
Comment (mark-solana) [reply core/src/blocktree.rs]: It doesn't, it uses `Borrow` and only uses a reference to blobs

**Comment by mark-solana** [comment]
@rob-solana given the issues I ran into with cluster info tests overflowing the stack, then I'll change that module to use `Box<Blob>` as well.

Or maybe we should just keep `SharedBlob`? We never actually use the sharing semantics of `Arc<RwLock<T>>` as far as I can tell, so there's probably only specific spots using it to avoid unnecessary copies. I know erasure was using `SharedBlob`s for no reason.

File: core/src/cluster_info.rs [hunk 2]
```diff
@@ -63,6 +63,9 @@ pub const GOSSIP_SLEEP_MILLIS: u64 = 100;
 /// the number of slots to respond with when responding to `Orphan` requests
 pub const MAX_ORPHAN_REPAIR_RESPONSES: usize = 10;
 
+/// Listen thread stack size set to 4MB
+const LISTEN_STACK_SIZE: usize = 4 * 1024 * 1024;
```
Comment (mark-solana) [inline core/src/cluster_info.rs]: Get rid of this and use `Box<Blob>`

File: core/src/retransmit_stage.rs [hunk 3]
```diff
@@ -6,7 +6,7 @@ use crate::cluster_info::{
     compute_retransmit_peers, ClusterInfo, DATA_PLANE_FANOUT, GROW_LAYER_CAPACITY,
     NEIGHBORHOOD_SIZE,
 };
-use crate::packet::SharedBlob;
+use crate::packet::Blob;
```
Comment (mark-solana) [inline core/src/retransmit_stage.rs]: Investigate if removing `SharedBlob` will cause unnecessary copies in this module

**Comment by mark-solana** [comment]
Not planning on working on this more in the immediate future, needs a lot of careful profiling. Closing for now.

**Pull request closed without merging**